### PR TITLE
Refactoring Internal Key generation and Adding Configs

### PR DESCRIPTION
**Pull request by rashm1n**
$ Subject
### Related Issue
* https://github.com/wso2-enterprise/choreo/issues/20341
### Description
* Refactoring Internal Key generation and Adding Configs

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java [hunk 1]
```diff
@@ -572,6 +573,18 @@ private void readChildElements(OMElement serverConfig,
                 setExtensionListenerConfigurations(element);
             } else if (APIConstants.JWT_AUDIENCES.equals(localName)){
                 setRestApiJWTAuthAudiences(element);
+            } else if ("APIKeyConfigs".equals(localName)) {
```
Comment (shanakama) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java:576]: move `APIKeyConfigs` string to `APIConstants` and refer.

Comment (rashm1n) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java]: Done.

**Review by shanakama — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java [hunk 2]
```diff
@@ -2966,4 +2966,10 @@ public static class ApiManagerConfigurationConstants {
 
     // Constants related to basic health check APIs
     public static final String WEB_SOCKET_HEALTH_CHECK_PATH = "/health";
+
+    // Constants related to internal API key configurations
+    public static final String INTERNAL_API_KEY_EXPIRY_TIME = "exp";
+    public static final String API_KEY_CONFIGURATIONS = "APIKeyConfigs";
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java:2972]:
```suggestion
    public static final String API_KEY_CONFIGURATIONS = "APITestKeyConfigs";
```

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2 [hunk 3]
```diff
@@ -1484,4 +1484,10 @@
         </Properties>
       </OperationPolicyConfigs>
       {% endif %}
+
+     {% if apim.testkey.validity is defined %}
+     <APIKeyConfigs>
+        <Exp>{{apim.testkey.validity.exp}}</Exp>
+     </APIKeyConfigs>
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2:1491]:
```suggestion
     <APITestKeyConfigs>
        <Exp>{{apim.testkey.validity.exp}}</Exp>
     </APITestKeyConfigs>
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 4]
```diff
@@ -3681,13 +3682,15 @@ public Response createNewAPIVersion(String newVersion, String apiId, Boolean def
 
     @Override
     public Response generateInternalAPIKey(String apiId, MessageContext messageContext) throws APIManagementException {
+        Map<String, String> configs = APIManagerConfiguration.getApiKeyProperties();
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:3685]:
```suggestion
        Map<String, String> configs = APIManagerConfiguration.getApiTestKeyProperties();
```

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**